src/clj/game/cards/assets.clj

```diff
@@ -1867,14 +1867,15 @@
                    (continue-ability
                      {:optional
                       {:prompt "Trash Rashida Jaheem to gain 3 [Credits] and draw 3 cards?"
+                       :msg "to trash itself"
                        :yes-ability
                        {:async true
-                        :msg "gain 3 [Credits] and draw 3 cards"
                         :effect (req (wait-for
                                        (trash state side card {:cause-card card})
                                        (wait-for
                                          (gain-credits state side 3)
-                                         (draw state side eid 3))))}}}
+                                         (draw state side eid 3)
+                                         (system-msg state side (str "uses " (:title card) " to gain 3 [Credits] and draw 3 cards")))))}}}
```

Comment:
Either wait-for the draw with a new eid and then call effect-completed on the existing eid, or call system-msg before draw.